
syntax = "proto3";

package unary;

service Integration{
    rpc RegisterIntegration(RegisterIntegrationParams) returns (RegisterIntegrationReturn) {}
    rpc GetIntegration(GetIntegrationParams) returns (GetIntegrationReturn) {}
    rpc StopIntegrationServer(StopIntegrationServerParams) returns (StopIntegrationServerReturn) {}
}

message RegisterIntegrationParams{
    string address = 1;
    int32 port = 2;
    repeated int32 ids = 3;
}

message RegisterIntegrationReturn{
    int32 numOfIds = 4;
}

message GetIntegrationParams{
    int32 id = 5;
}

message GetIntegrationReturn{
    string address = 6;
    int32 port = 7;
}

message StopIntegrationServerParams{}

message StopIntegrationServerReturn{
    int32 numOfIds = 8;
}